### PR TITLE
IDENTITY-6113: SSO Agent fixes identified during QSG SSO SAML demo app implementations

### PR DESCRIPTION
**Pull request by madumalt**
During a Quick Start Guide SSO Demo App development found theses minor issues and fixed them.
Some comments were introduced to clarify things with the intention of avoiding possible misleads.

### CODE REVIEW

**Comment by madumalt** [comment]
@pulasthi7 @isharak @sagara-gunathunga @kasungayan

**Comment by madumalt** [comment]
IDENTITY-6113

File: components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SAML2SSOManager.java [hunk 1]
```diff
@@ -468,14 +468,15 @@ protected void processSSOResponse(HttpServletRequest request) throws SSOAgentExc
             throw new SSOAgentException("SAML2 Response does not contain the name of the subject");
         }
 
-
+        // This should be the only time where a new session can be created.
+        // Thus in latter palces getsession(false) should be used.
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SAML2SSOManager.java]: typo: palces

File: components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SSOAgentHttpSessionListener.java [hunk 2]
```diff
@@ -34,6 +34,9 @@
     @Override
     public void sessionCreated(HttpSessionEvent httpSessionEvent) {
         if (httpSessionEvent.getSession().getAttribute(SSOAgentConstants.SESSION_BEAN_NAME) == null) {
+            // This log is not accurate, since we depend on .getsessio() to create new session
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SSOAgentHttpSessionListener.java]: typo: .getsessio()

File: components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SAML2SSOManager.java (hunk 1 above)
Comment (mefarazath) [inline components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SAML2SSOManager.java]: let's say request.getSession(false)

Comment (madumalt) [inline components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SAML2SSOManager.java]: done

File: components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SSOAgentHttpSessionListener.java (hunk 2 above)
Comment (madumalt) [inline components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SSOAgentHttpSessionListener.java]: done

**Comment by madumalt** [comment]
Updated the PR.
Please review and merge.

**Review by mefarazath — APPROVED** [review]

**Pull request merged**